WISE-Community/WISE
# Add license and authors info to project metadata

Add `wise.hostname` to wise.properties file. For example:
```
wise.hostname=http://localhost:8080
# wise.hostname=https://wise.berkeley.edu
```

To test:
- Log in as a teacher and create a new project using the authoring tool.
  - Verify that `license.txt` file has been created in the project folder and that the `authors` field has been added to the project metadata (both in db and `project.json`).
  - Make sure `license.txt` file and `authors` field match the examples in #1261.
- Copy the project you just created using three different methods: 1) copy from site (`/teacher/library/personal`), 2) set up a run with the project, 3) copy from authoring tool.
  - Verify `license.txt` is generated correctly for copied projects and that the `parentProject` field is added correctly to the project metadata (see example in #1261).
  - Make sure that the project metadata's `authors` field is set to an empty array.
- Edit one of the copied projects using the authoring tool.
  - Make sure that the logged in user is added to the project metadata's `authors` field and that the `license.txt` file is updated with the new author name.
- Edit a project's title.
  - Make sure that the `license.txt` file is updated with the new title.
- Share one of the projects with a different teacher. Log in as that teacher and edit the shared project in the authoring tool.
  - Make sure that the new teacher is added to the metadata `authors` field and that the `license.txt` file is updated with the new author's name.
- Export a project and then import it using the admin tools.
  - Check that the `uri` field is added to the exported project's metadata.
  - Make sure the `license.txt` is generated correctly (the imported project should appear as a copy of the source project) and that the `parentProject` field is added to the new project's metadata.
  - Make sure the `id` fields for the project and parent authors are removed from the `parentProject` object (since the imported project might have come from another WISE instance).
  - Check that the `authors` metadata field in the new project is an empty array.
  - Check that the `uri` field in the new project has been updated correctly (and doesn't show the source project's preview link)

Resolves #1261.
